src/Modal/ModalPopup.tsx

```diff
@@ -1,27 +1,71 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import { FocusOn } from 'react-focus-on';
+import type { Placement } from '@popperjs/core';
+import { Modifier } from 'react-popper';
 import Portal from './Portal';
-import PopperElement from './PopperElement';
 import { ModalContextProvider } from './ModalContext';
+import PopperElement, { PopperElementProps } from './PopperElement';
 
-const PLACEMENT_OFFSETS = { right: [-2, 10], left: [-2, 10] };
+type PlacementOffsets = {
+  [K in Placement]: number[] | undefined;
+};
+
+const PLACEMENT_OFFSETS: PlacementOffsets = {
+  right: [-2, 10],
+  left: [-2, 10],
+  auto: undefined,
+  'auto-start': undefined,
+  'auto-end': undefined,
+  top: undefined,
+  bottom: undefined,
+  'top-start': undefined,
+  'top-end': undefined,
+  'bottom-start': undefined,
+  'bottom-end': undefined,
+  'right-start': undefined,
+  'right-end': undefined,
+  'left-start': undefined,
+  'left-end': undefined,
+};
+
+interface ModalPopupProps extends PopperElementProps {
+  /** Specifies the contents of the modal */
+  children: React.ReactNode,
+  /** A callback function for when the modal is dismissed */
+  onClose: () => void,
+  /** Is the modal dialog open or closed */
+  isOpen: boolean,
+  /** Prevent clicking on the backdrop or pressing Esc to close the modal */
```

Comment:
```suggestion
  /** Prevent clicking on the backdrop or pressing Esc to close the modal. Default: false */
```
Please mention the default values (if any) in the docstrings, because otherwise they don't show up on the docs site after conversion to TypeScript.

Reply:
The docs display default values nicely if we include them in the function call. 488ffb828fe8

<img width="503" height="333" alt="image" src="https://github.com/user-attachments/assets/46ea08c6-a4b6-476f-9865-01b58093a1f8" />
kk

Reply:
Oh nice. I guess I didn't do that in cases where bootstrap specified the default, but that does seem better.